aws/src/main/java/org/apache/iceberg/aws/glue/LockManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * An interface for locking, used to ensure Glue catalog commit isolation.
+ */
+interface LockManager extends AutoCloseable {
+
+  /**
+   * Try to acquire a lock
+   * @param lockId lock ID
+   * @throws LockNotAcquiredException if lock is not acquired
+   */
+  void acquire(LockId lockId);
```

Comment:
Why not use `lock` and `unlock`?

Reply:
Not a significant reason at all, but otherwise the config key would be like `lock.lock-timeout-ms` which has 2 `lock`s...
I am okay with either lock & unlock or acquire & release.